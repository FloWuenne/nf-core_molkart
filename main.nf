#!/usr/bin/env nextflow
/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    nf-core/molkart
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Github : https://github.com/nf-core/molkart
    Website: https://nf-co.re/molkart
    Slack  : https://nfcore.slack.com/channels/molkart
----------------------------------------------------------------------------------------
*/

nextflow.enable.dsl = 2

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    GENOME PARAMETER VALUES
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

params.fasta = WorkflowMain.getGenomeAttribute(params, 'fasta')

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    VALIDATE & PRINT PARAMETER SUMMARY
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

WorkflowMain.initialise(workflow, params, log)

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    NAMED WORKFLOW FOR PIPELINE
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

include { MOLKART } from './workflows/molkart'

//
// WORKFLOW: Run main nf-core/molkart analysis pipeline
//
workflow NFCORE_MOLKART {
    MOLKART ()
}

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    RUN ALL WORKFLOWS
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

//
// WORKFLOW: Execute a single named workflow for the pipeline
// See: https://github.com/nf-core/rnaseq/issues/619
//
workflow {
    NFCORE_MOLKART ()
}

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    THE END
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/
