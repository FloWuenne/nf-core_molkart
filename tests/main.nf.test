nextflow_pipeline {

    name "Test Workflow main.nf"
    script "../main.nf"
    config "./nextflow.config"
    tag "pipeline"
    tag "pipeline_molkart"

    test("Nuclear channel, mesmer and cellpose, without clahe") {

        when {
            params {
                input = 'https://raw.githubusercontent.com/nf-core/test-datasets/molkart/test_data/samplesheets/samplesheet_nuclear.csv'
                outdir = "$outputDir"
                skip_clahe = true
                tilesize = 90
                segmentation_method = "cellpose,mesmer"
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                path("$outputDir/mindagap/nuclear_gridfilled.tiff"),
                path("$outputDir/mindagap/spots_markedDups.txt"),
                path("$outputDir/segmentation/cellpose/nuc_only_cellpose_mask.tif"),
                path("$outputDir/segmentation/mesmer/nuc_only_mesmer_mask.tif"),
                path("$outputDir/spot2cell/spots_markedDups.cellpose.cellxgene.csv"),
                path("$outputDir/spot2cell/spots_markedDups.mesmer.cellxgene.csv"),
                path("$outputDir/molcart_qc/nuc_only.cellpose.spot_QC.csv"),
                path("$outputDir/molcart_qc/nuc_only.mesmer.spot_QC.csv")
            ).match()
            assert file("$outputDir/multiqc/multiqc_report.html").exists()
        }
    }

    test("Two channels, mesmer and cellpose, with clahe") {

        when {
            params {
                input = 'https://raw.githubusercontent.com/nf-core/test-datasets/molkart/test_data/samplesheets/samplesheet_membrane.csv'
                outdir = "$outputDir"
                tilesize = 90
                clahe_skip_pyramid = true
                segmentation_method = "cellpose,mesmer"
                }
        }

        then {
            assert workflow.success
            assert snapshot(
                path("$outputDir/mindagap/membrane_gridfilled.tiff"),
                path("$outputDir/mindagap/spots_markedDups.txt"),
                path("$outputDir/segmentation/cellpose/mem_only_cellpose_mask.tif"),
                path("$outputDir/segmentation/mesmer/mem_only_mesmer_mask.tif"),
                path("$outputDir/spot2cell/spots_markedDups.cellpose.cellxgene.csv"),
                path("$outputDir/spot2cell/spots_markedDups.mesmer.cellxgene.csv"),
                path("$outputDir/stack/mem_only.stack.ome.tif"),
                path("$outputDir/molcart_qc/mem_only.cellpose.spot_QC.csv"),
                path("$outputDir/molcart_qc/mem_only.mesmer.spot_QC.csv"),
            ).match()
            assert file("$outputDir/clahe/mem_only_0.clahe.tiff").exists()
            assert file("$outputDir/clahe/mem_only_1.clahe.tiff").exists()
            assert file("$outputDir/multiqc/multiqc_report.html").exists()
        }
    }

    test("Create training subset") {

        when {
            params {
                input = 'https://raw.githubusercontent.com/nf-core/test-datasets/molkart/test_data/samplesheets/samplesheet_nuclear.csv'
                outdir = "$outputDir"
                tilesize = 90
                clahe_skip_pyramid = true
                create_training_subset = true
                crop_amount = 2
                crop_size_x = 10
                crop_size_y = 10
                }
        }

        then {
            assert workflow.success
            assert snapshot(
                path("$outputDir/mindagap/nuclear_gridfilled.tiff"),
            ).match()
            assert file("$outputDir/clahe/nuc_only_0.clahe.tiff").exists()
            assert file("$outputDir/training_subset/hdf5/nuc_only_0.clahe_crop0.hdf5").exists()
            assert file("$outputDir/training_subset/hdf5/nuc_only_0.clahe_crop1.hdf5").exists()
            assert file("$outputDir/training_subset/hdf5/nuc_only_0.clahe_CropSummary.txt").exists()
            assert file("$outputDir/training_subset/tiff/nuc_only_0.clahe_crop0.tiff").exists()
            assert file("$outputDir/training_subset/tiff/nuc_only_0.clahe_crop1.tiff").exists()
            assert file("$outputDir/training_subset/tiff/nuc_only_0.clahe.crop_overview.png").exists()
        }
    }

}
