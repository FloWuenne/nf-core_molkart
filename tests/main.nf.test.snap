{
    "Two channels, mesmer and cellpose, with clahe": {
        "content": [
            "mem_only_membrane_gridfilled.tiff:md5,6e24160d758468d2de07ca200dfb62c2",
            "mem_only_nuclear_gridfilled.tiff:md5,123763d54f05b2274690b3b84f9690b1",
            "mem_only_spots_markedDups.txt:md5,4562caad05850d7dd7b6e9235e068a8b",
            "mem_only_cellpose_mask.tif:md5,590591f541b6d3f17810cd44dd71b252",
            "mem_only_mesmer_mask.tif:md5,6f8c792ccc4cca97b5696990fb4e3801",
            "mem_only_mesmer_filtered.tif:md5,5751b2e39393a9c8221396f5ef592035",
            "mem_only_cellpose_filtered.tif:md5,590591f541b6d3f17810cd44dd71b252",
            "cellxgene_mem_only_cellpose.csv:md5,f063c1ce5e93e1e73431af36c6bc1e79",
            "cellxgene_mem_only_mesmer.csv:md5,198b187d5f151077437aa591f5991b22",
            "mem_only_cellpose.adata:md5,b47ea63ffea0947e43511f17b6920cd8",
            "mem_only_mesmer.adata:md5,530649647a3466316ea52dde9dede4ab",
            "mem_only.cellpose.spot_QC.csv:md5,df312175498f7942bffc33f2c2d8d1c9",
            "mem_only.mesmer.spot_QC.csv:md5,3273bd6fecf93a3d240614d1b38831c9"
        ],
        "timestamp": "2024-01-05T11:19:49.882138591"
    },
    "Nuclear channel, mesmer and cellpose, without clahe": {
        "content": [
            "nuc_only_nuclear_gridfilled.tiff:md5,123763d54f05b2274690b3b84f9690b1",
            "nuc_only_spots_markedDups.txt:md5,4562caad05850d7dd7b6e9235e068a8b",
            "nuc_only_cellpose_mask.tif:md5,2784d8c6683ad80c24d8df4121e1128f",
            "nuc_only_mesmer_mask.tif:md5,e9719f53651b4bf76a37e9374db7f4e4",
            "nuc_only_mesmer_filtered.tif:md5,5a6761c80ea65d41ce00d06d3414c96b",
            "nuc_only_cellpose_filtered.tif:md5,2784d8c6683ad80c24d8df4121e1128f",
            "cellxgene_nuc_only_cellpose.csv:md5,0610a8713457c28acf1bc4c298bee8e3",
            "cellxgene_nuc_only_mesmer.csv:md5,425cfb90a60b564e4bfff09ac5ea94c7",
            "nuc_only_cellpose.adata:md5,fd52e62711465d754fd36a433761cb3b",
            "nuc_only_mesmer.adata:md5,cd20ab6db5274bb85c960ea3bd8d2619",
            "nuc_only.cellpose.spot_QC.csv:md5,e77b5973e0997170d0fde5c5901ad551",
            "nuc_only.mesmer.spot_QC.csv:md5,f8f4eb85bb8269341ac072ac78962ed4"
        ],
        "timestamp": "2024-01-05T11:18:48.214995389"
    },
    "Create training subset": {
        "content": [
            "nuc_only_nuclear_gridfilled.tiff:md5,123763d54f05b2274690b3b84f9690b1"
        ],
        "timestamp": "2023-11-28T21:40:40.925034909"
    }
}